File: leetCode/array/Find_Pivot_Index_724/Main.java
import java.util.Arrays;

class Solution {
    public int pivotIndexN(int[] nums) {
        if (nums.length == 0) return -1;
        int[] sumsArr = new int[nums.length + 1];
        Arrays.fill(sumsArr, 0);
        for(int i = 0; i < nums.length; i++) {
            sumsArr[i + 1] = sumsArr[i] + nums[i];
        }
        for (int i = 0; i < nums.length; i++) {
            int left = sumsArr[i - 1];
            int right = sumsArr[sumsArr.length - 1] - left - nums[i];
            if(left == right) {
                return i;
            }
        }
        return -1;
    }

    public int pivotIndex(int[] nums) {
        if (nums.length == 0) return -1;
        int sum = 0;
        int left = 0;
        for(int i = 0; i < nums.length; i++) {
            sum += nums[i];
        }
        for (int i = 0; i < nums.length; i++) {
            int right = sum - left - nums[i];
            if(left == right) return i;
            left += nums[i];
        }
        return -1;
    }
}

public class Main {
    public static void main(String[] args) {
        Solution s = new Solution();
        int[] a = {-1,-1,-1,-1,0,1};
        int res = s.pivotIndex(a);
        System.out.println(res);
    }
}

File: leetCode/array/Find_Pivot_Index_724/main.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        arr_sum = sum(nums)

        left_sum = 0
        for i, num in enumerate(nums):
            right_sum = arr_sum - num - left_sum
            if right_sum == left_sum:
                return i
            left_sum += num
        return -1

    def pivotIndex_slow(self, nums: List[int]) -> int:
        l_sums = [0]
        for i in range(0, len(nums) - 1):
            l_sums.append(l_sums[i] + nums[i])

        for i in range(len(nums)):
            r_sum = sum(nums[i + 1:])
            if l_sums[i] == r_sum:
                return i

        return -1
